esphome/components/ethernet/ethernet_component.cpp

```diff
@@ -234,6 +234,22 @@ void EthernetComponent::setup() {
   ESPHL_ERROR_CHECK(err, "GOT IPv6 event handler register error");
 #endif /* USE_NETWORK_IPV6 */
 
+  if (this->duplex_mode_.has_value() || this->link_speed_.has_value()) {
+    bool auto_nego_en = false;
+    err = esp_eth_ioctl(this->eth_handle_, ETH_CMD_S_AUTONEGO, &auto_nego_en);
+    ESPHL_ERROR_CHECK(err, "ETH_CMD_S_AUTONEGO error");
+
+    if (this->duplex_mode_.has_value()) {
+      esp_eth_ioctl(this->eth_handle_, ETH_CMD_S_DUPLEX_MODE, &this->duplex_mode_);
```

Comment:
```suggestion
      err = esp_eth_ioctl(this->eth_handle_, ETH_CMD_S_DUPLEX_MODE, &this->duplex_mode_.value());
```

This doesn't look like its checkin the err value?